src/OrchardCore.Modules/OrchardCore.Media/Services/ChunkFileUploadService.cs

```diff
@@ -0,0 +1,262 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Modules;
+
+namespace OrchardCore.Media.Services;
+
+public class ChunkFileUploadService : IChunkFileUploadService
```

Comment:
I think we need to rethink about this because some media providers already have support upload chunks for example Azure

/cc @Piedone

Reply:
The OC application is a proxy between the user's browser and the Azure storage, so the user doesn't upload the media file to Azure directly, first, the OC application receives and forwards it to Azure. See `IFileStorage.CreateFileFromStreamAsync` implementation:

https://github.com/Lombiq/OrchardCore/blob/b5a9ebb780b8fa8260d7aaf201358f47d3464727/src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs#L371-L401

It is called via the `IMediaFileStore.CreateFileFromStreamAsync` in the `OrchardCore.Media.Controller.AdminController.Upload`

https://github.com/Lombiq/OrchardCore/blob/b5a9ebb780b8fa8260d7aaf201358f47d3464727/src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs#L193

So I think all media files go through the `OrchardCore.Media.Controller.AdminController.Upload` independently of the applied storage provider.